### PR TITLE
Fix service status in relation data

### PR DESCRIPTION
**Pull request by Gmerold**
- Fixes setting service status passed in relation data in libs
- Fixes setting service status passed in relation data in charms not using libs
- Unifies relations handling across the charms
- Minor code and unit tests improvements

NOTE: This PR's primary goal is to update the libs. In the next step, charms will be updated with the latest libs.

### CODE REVIEW

File: orchestrator-bundle/nms-magmalte-operator/tests/unit/test_charm.py [hunk 1]
```diff
@@ -108,19 +119,18 @@ def test_given_pod_is_leader_when_database_relation_joined_event_then_database_i
 
         self.assertEqual(db_event.database, self.TEST_DB_NAME)
 
-    @patch("ops.model.Container.exec")
     @patch("charm.MagmaNmsMagmalteCharm._get_db_connection_string", new_callable=PropertyMock)
-    @patch("charm.MagmaNmsMagmalteCharm._domain_name", new_callable=PropertyMock)
+    @patch("charm.MagmaNmsMagmalteCharm._create_master_nms_admin_user", Mock())
```
Comment (gruyaume) [inline orchestrator-bundle/nms-magmalte-operator/tests/unit/test_charm.py:123]: If I'm not wrong you don't have to specify it to be a "Mock" object, it is that by default.

File: orchestrator-bundle/orc8r-orchestrator-operator/src/charm.py [hunk 2]
```diff
@@ -228,39 +230,60 @@ def _set_log_verbosity_action(self, event: ActionEvent):
...
+        return True
+
     @property
     def _relations_ready(self) -> bool:
         """Checks whether required relations are ready."""
-        required_relations = ["magma-orc8r-certifier", "metrics-endpoint"]
-        missing_relations = [
-            relation for relation in required_relations if not self._relation_active(relation)
+        not_ready_relations = [
+            relation for relation in self.REQUIRED_RELATIONS if not self._relation_active(relation)
```
Comment (gruyaume) [inline orchestrator-bundle/orc8r-orchestrator-operator/src/charm.py:261]: Does it make sense ot check if the "metrics-endpoint" relation is active?

File: orchestrator-bundle/orc8r-orchestrator-operator/tests/unit/test_charm.py [hunk 3]
```diff
@@ -172,6 +173,65 @@ def test_given_bad_elasticsearch_config_when_on_config_changed_event_then_status
             "Config for elasticsearch is not valid. Format should be <hostname>:<port>"
         )
 
+    @patch("charm.MagmaOrc8rOrchestratorCharm._mount_certifier_certs")
+    @patch(
+        "charm.MagmaOrc8rOrchestratorCharm._nms_certs_mounted", PropertyMock(return_value=False)
+    )
+    def test_given_charm_with_certifier_relation_active_when_certs_are_not_mounted_then_mount_orc8r_certs(  # noqa: E501
```
Comment (gruyaume) [inline orchestrator-bundle/orc8r-orchestrator-operator/tests/unit/test_charm.py]: Drop the `charm` word. It is obvious that we are talking about charms.

Comment (gruyaume) [reply orchestrator-bundle/orc8r-orchestrator-operator/tests/unit/test_charm.py]: Also, the `then` statement doesn't match with the actual test content

File: orchestrator-bundle/orc8r-orchestrator-operator/tests/unit/test_charm.py [hunk 4]
```diff
@@ -172,6 +173,65 @@ def test_given_bad_elasticsearch_config_when_on_config_changed_event_then_status
...
+        self.harness.add_relation_unit(relation_id, "orc8r-certifier/0")
+        self.harness.update_relation_data(relation_id, "orc8r-certifier/0", {"active": "True"})
+
+        mock_on_certifier_relation_changed.assert_not_called()
+
+    @patch("charm.MagmaOrc8rOrchestratorCharm._namespace", PropertyMock(return_value="qwerty"))
+    @patch("charm.MagmaOrc8rOrchestratorCharm._relations_ready", PropertyMock(return_value=True))
+    @patch("charm.MagmaOrc8rOrchestratorCharm._relations_created", PropertyMock(return_value=True))
+    @patch("charm.MagmaOrc8rOrchestratorCharm._nms_certs_mounted", PropertyMock(return_value=True))
+    def test_given_magma_orc8r_orchestrator_service_running_when_metrics_magma_orc8r_orchestrator_relation_joined_event_emitted_then_active_key_in_relation_data_is_set_to_true(  # noqa: E501
```
Comment (gruyaume) [inline orchestrator-bundle/orc8r-orchestrator-operator/tests/unit/test_charm.py]: when metrocs magma orc8r orchestrator relation joined?

File: orchestrator-bundle/orc8r-orchestrator-operator/tests/unit/test_charm.py [hunk 5]
```diff
@@ -172,6 +173,65 @@ def test_given_bad_elasticsearch_config_when_on_config_changed_event_then_status
...
+    ):
+        relation_id = self.harness.add_relation("magma-orc8r-certifier", "orc8r-certifier")
+        self.harness.add_relation_unit(relation_id, "orc8r-certifier/0")
+        self.harness.update_relation_data(relation_id, "orc8r-certifier/0", {"active": "True"})
+
+        mock_on_certifier_relation_changed.assert_called_once()
+
+    @patch("charm.MagmaOrc8rOrchestratorCharm._mount_certifier_certs")
+    @patch("charm.MagmaOrc8rOrchestratorCharm._nms_certs_mounted", PropertyMock(return_value=True))
+    def test_given_charm_with_certifier_relation_active_when_certs_are_mounted_then_dont_mount_orc8r_certs(  # noqa: E501
```
Comment (gruyaume) [inline orchestrator-bundle/orc8r-orchestrator-operator/tests/unit/test_charm.py]: `then` statement doesn't match the test assert section.

File: orchestrator-bundle/nms-magmalte-operator/tests/unit/test_charm.py (hunk 1 above)
Comment (Gmerold) [reply orchestrator-bundle/nms-magmalte-operator/tests/unit/test_charm.py:123]: If you don't specify it, you need to have a corresponding function argument. Specifying it as a mock objects, allows you to omit that.

File: orchestrator-bundle/orc8r-orchestrator-operator/src/charm.py (hunk 2 above)
Comment (Gmerold) [reply orchestrator-bundle/orc8r-orchestrator-operator/src/charm.py:261]: `metrics-endpoint` may not be the most fortunate name, but in fact, it indicates readiness of the `prometheus-edge-hub`, which is required to start before `orc8r-orchestrator`.

File: orchestrator-bundle/orc8r-orchestrator-operator/tests/unit/test_charm.py (hunk 3 above)
Comment (Gmerold) [reply orchestrator-bundle/orc8r-orchestrator-operator/tests/unit/test_charm.py]: Fixed.

File: orchestrator-bundle/orc8r-orchestrator-operator/tests/unit/test_charm.py (hunk 5 above)
Comment (Gmerold) [reply orchestrator-bundle/orc8r-orchestrator-operator/tests/unit/test_charm.py]: Fixed.

File: orchestrator-bundle/orc8r-orchestrator-operator/tests/unit/test_charm.py (hunk 4 above)
Comment (Gmerold) [reply orchestrator-bundle/orc8r-orchestrator-operator/tests/unit/test_charm.py]: Looks like copy-pasting error ;)
Fixed.

File: orchestrator-bundle/orc8r-orchestrator-operator/src/charm.py (hunk 2 above)
Comment (gruyaume) [reply orchestrator-bundle/orc8r-orchestrator-operator/src/charm.py:261]: But does the prometheus-edge-hub charm does like the other magma charm and provide the state of its workload service?

Comment (Gmerold) [reply orchestrator-bundle/orc8r-orchestrator-operator/src/charm.py:261]: It does :) You even approved the PR for that change ;)

**Review by gruyaume — APPROVED** [review]

**Pull request merged**